frontend/src/components/modals/CopyLink.vue

```diff
@@ -0,0 +1,132 @@
+<template>
+  <div v-if="isCopyLinkModalVisible"
+       class="modal-backdrop"
+       tabindex="-1"
+       role="dialog"
+       aria-labelledby="copylinkModalLabel"
+       aria-hidden="true"
+       @keypress.enter="copylink"
+       @click="close"
+  >
+    <div class="modal-dialog modal-dialog-centered modal-sm" role="document">
+      <div class="modal-content container-fluid" @click.stop="">
+        <div class="modal-header">
+          <h5 class="modal-title font-weight-bold">Copy Link</h5>
+          <button type="button"
+                  class="close"
+                  id="closeButton"
+                  data-toggle="tooltip"
+                  data-placement="right"
+                  title="Close"
+                  aria-label="Close"
+                  @click="close"
+          >
+            <span aria-hidden="true">&times;</span>
+          </button>
+        </div>
+        <div class="modal-body">
+          <div class="form-group">
+            <div class="input-group">
+              <input v-model="itemName"
+                     type="text"
+                     class="form-control"
+                     ref="inputField"
+                     placeholder="Link"
+              >
+            </div>
+          </div>
+        </div>
+        <div class="modal-footer">
+          <button type="button" class="btn font-weight-bold" id="copyButton" @click="copylink">
+            Select text
+          </button>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { parseXML } from 'webdav'
+
+  export default {
+    name: 'CopyLinkModal',
+    props: ["isCopyLinkModalVisible"],
+    data() {
+      return {
+        itemName: "",
+      }
+    },
+    async updated() {
+      if (this.isCopyLinkModalVisible) {
```

Comment:
Here you can also add an event listener with escape handler, so that the modal is also closable when Esc is pressed. You can copy this behavior from any of my modals.